Code/Framework/AzCore/AzCore/Memory/AllocationRecords.cpp

```diff
@@ -322,20 +325,26 @@ namespace AZ::Debug
 #endif
 
         // delete allocation record
-        if (allocationInfo.m_namesBlock)
-        {
-            m_records.get_allocator().deallocate(allocationInfo.m_namesBlock, allocationInfo.m_namesBlockSize, 1);
-            allocationInfo.m_namesBlock = nullptr;
-            allocationInfo.m_namesBlockSize = 0;
-            allocationInfo.m_name = nullptr;
-            allocationInfo.m_fileName = nullptr;
-        }
-        if (allocationInfo.m_stackFrames)
+#if defined(CARBONATED)
+        if (!m_enumerationInProgress)  // intentional memory leak here to provide the right data for enumeration callback
```

Comment:
This is not 100% guarantee again an occasional deletion, but 99.999 I guess, I seen no issues during testing.
Windows ignores invalid addresses, but libunwind crashes. This fixes the crashes, but libunwind still does not work right, it cannot even read the call stack, there is something else messing it.